lua/better-breachable-doors/sv_breach.lua

```diff
@@ -0,0 +1,757 @@
+    end
+
+    -- Apply damage
+    local healthAfterDamage = healthWithRegen - damageToTake
+
+    -- Update the door's Data Table Network Variables
+    door:SetIsHandleDamage( isHandleDamage )
+    door:SetHealthAfterLastDamage( healthAfterDamage )
+    door:SetDamageTime( time )
+    door:SetDamageDirection( BBD.GetDoorOpenDirection( door, dmg ) )
+
+    if healthAfterDamage <= 0 then
+        BBD.OnDoorBreached( door, dmg )
+    else
+        BBD.PlayDamageSound( door )
+    end
+end
+
+--#endregion Door Damage Logic
+
+--#region Callbacks
+
+BBD.UpdateMaxHealths = function( _, oldMaxHealth, newMaxHealth )
+    for _, door in pairs( ents.FindByClass( "prop_door_rotating" ) ) do
+        if not IsValid( door ) then continue end
+
+        local oldHealth = door:GetHealthAfterLastDamage()
+        local wasMaxHealth = oldHealth == tonumber( oldMaxHealth )
+
+        -- Purely for variable name clarity
+        local newHealth = oldHealth
+
+        -- Keep full-health doors at full health
+        if wasMaxHealth then
+            newHealth = newMaxHealth
+        end
+
+        -- Don't let doors have more health than the new max
+        newHealth = math.min( newHealth, tonumber( newMaxHealth ) )
+
+        door:SetHealthAfterLastDamage( newHealth )
+    end
+end
+
+BBD.OnDoorUsed = function( ply, door )
+    if not door or not IsValid( door ) then return end
+    if door:GetClass() ~= "prop_door_rotating" then return end
+
+    -- If the door is breached, don't allow interaction
+    if door:GetHealthAfterLastDamage() <= 0 then
+        return false
+    end
+
+    -- If any connected doors are breached, don't allow interaction
+    if BBD.DoorHasConnections( door ) then
+        for _, connectedDoor in pairs( BBD.GetConnectedDoors( door ) ) do
+            if not IsValid( connectedDoor ) then continue end
+
+            if connectedDoor:GetHealthAfterLastDamage() <= 0 then
+                return false
+            end
+        end
+    end
+end
+
+--#endregion Callbacks
+
+local hotloaded = false
+if BBD.Enable then hotloaded = true BBD.Disable() end
+
+-- Enable the door breach system
+BBD.Enable = function()
+    -- Tracking door damage
+    hook.Add( "EntityTakeDamage", BBD.HOOK_DAMAGE_DETECTION, BBD.OnDoorDamaged )
+
+    -- Check for player's colliding with respawning doors
+    hook.Add( "Think", BBD.HOOK_CHECK_COLLISIONS, BBD.CheckPlayerCollisions )
+
+    -- Don't allow players to use breached doors
+    hook.Add( "PlayerUse", BBD.HOOK_SUPPRESS_USE, BBD.OnDoorUsed )
+
+    -- Update the max health of all doors when the ConVar changes
+    cvars.AddChangeCallback( conVarMaxHealth:GetName(), BBD.UpdateMaxHealths, BBD.CONVAR_CALLBACK_HEALTH )
+
+    -- Ensure all doors are marked as usable to DarkRP's prop protection system
+    -- Otherwise, the PlayerUse hook will not be called
+    for _, door in pairs( ents.FindByClass( "prop_door_rotating" ) ) do
+        door.PlayerUse = true
```

Comment:
Would you maybe want to check to make sure the existing value isn't `false`? Sounds like it would complicate the rest of your code though, and I'm not sure how useful it is